### ISSUE TITLE
Refactor setup.py to make it install the required packages

### ISSUE THREAD

**Issue by ndilalla** [lucabaldini/ixpeobssim#632]
Right now, running ``pip install .`` or the equivalent ``python setup.py`` on a clean environment causes the following error:
```
Traceback (most recent call last):
  File "/home/ndilalla/work/ixpe/ixpeobssim/setup.py", line 24, in <module>
    from ixpeobssim import PACKAGE_NAME
  File "/home/ndilalla/work/ixpe/ixpeobssim/ixpeobssim/__init__.py", line 28, in <module>
    import numpy
ModuleNotFoundError: No module named 'numpy'
```
The expected result should be the automatic installation of the required packages listed in the ``setup.py`` file, namely ``numpy``, ``matplotlib``, ``astropy``, ``scipy``, ``regions`` and ``skyfield``. This is happening because we try to import ``ixpeobssim`` stuff before the actual setup. The relevant lines of code are the following, defined in the ``setup.py``:
```
from ixpeobssim import PACKAGE_NAME
from ixpeobssim.__version__ import TAG
```
The first import can be easily avoided by defining the package name in the setup file, as this does not change. Getting the version is more complicated. My suggestion would be to use [``versioneer``](https://github.com/python-versioneer/python-versioneer) which requires to add a couple of files to the repository. In this way we can simply run ``versioneer.get_version()`` and get the version instead of importing the whole package. Comments?

**Comment by lucabaldini** [comment]
I don't think I understand all the implications, but please go ahead!

Is this pointing to something deeper that is not properly done and requires more extensive modifications?

**Comment by ndilalla** [comment]
Using `versioneer` has many advantages, including a proper integration with git which saves you from manually updating the version file during the tagging process. Unfortunately, using this tool will require some modifications in the package, most notably switching from a plain `__version__.py` file to a much more complex `_version.py` file, supported by `versioneer`. The alternative would be to manually read the `__version__.py` file without importing ixpeobssim. Let's think about it.

**Comment by lucabaldini** [comment]
Looking again at the code it looks to me that the issue is all the code parsing the versions of the various packages is living in the top-level init.py.

How about creating a utils.environ module in which we move all the versioning stuff requiring the import of third-party modules? Propagating that to the rest of the code would be trivial, as there is literally a handful of places where we actually use the information.

```
[lbaldini@pclbaldini ixpeobssim]$ grep PY_VERSION -r ixpeobssim/
ixpeobssim/core/fitting.py:from ixpeobssim import SCIPY_VERSION
ixpeobssim/core/fitting.py:    if SCIPY_VERSION < '0.18':
ixpeobssim/__init__.py:NUMPY_VERSION = retrieve_version(numpy)
ixpeobssim/__init__.py:SCIPY_VERSION = retrieve_version(scipy)
ixpeobssim/__init__.py:ASTROPY_VERSION = retrieve_version(astropy)
ixpeobssim/srcmodel/polarization.py:from ixpeobssim import ASTROPY_VERSION
ixpeobssim/srcmodel/polarization.py:        if ASTROPY_VERSION <= '3.1':
[lbaldini@pclbaldini ixpeobssim]$ grep MATPLOTLIB_VERSION -r ixpeobssim/
ixpeobssim/__init__.py:MATPLOTLIB_VERSION = retrieve_version(matplotlib)
[lbaldini@pclbaldini ixpeobssim]$ grep SKYFIELD_VERSION -r ixpeobssim/
ixpeobssim/__init__.py:SKYFIELD_VERSION = retrieve_version(skyfield)
ixpeobssim/instrument/traj.py:from ixpeobssim import IXPEOBSSIM_INSTRUMENT_DATA, IXPEOBSSIM_DATA, SKYFIELD_VERSION
ixpeobssim/instrument/traj.py:            if SKYFIELD_VERSION < '1.26':
ixpeobssim/instrument/traj.py:            if SKYFIELD_VERSION < '1.26':
ixpeobssim/instrument/traj.py:        if SKYFIELD_VERSION < '1.23':
[lbaldini@pclbaldini ixpeobssim]$ grep REGIONS_VERSION -r ixpeobssim/
ixpeobssim/utils/astro.py:from ixpeobssim import REGIONS_VERSION
ixpeobssim/utils/astro.py:    if REGIONS_VERSION <= '0.5':
ixpeobssim/__init__.py:REGIONS_VERSION = retrieve_version(regions)
[lbaldini@pclbaldini ixpeobssim]$ grep PYTHON_VERSION -r ixpeobssim/
ixpeobssim/__init__.py:PYTHON_VERSION = xPackageVersion(sys.version_info.major, sys.version_info.minor,
[lbaldini@pclbaldini ixpeobssim]$ 
``` 

And then, moving along, we can discuss later whether we want to switch to versioneer.

If this looks like a plan I can make the change today.

**Comment by ndilalla** [comment]
I like this solution!

**Comment by lucabaldini** [comment]
Pull request: https://github.com/lucabaldini/ixpeobssim/pull/633

@ndilalla Take a look and feel free to merge it---or I can do it as soon as you give me the good to go.

**Comment by lucabaldini** [comment]
Shipped in 29.1.2
